clmm/dataops/__init__.py

```diff
@@ -164,6 +165,143 @@ def compute_tangential_and_cross_components(
+    (2003), arXiv:astro-ph/0312036:
+
+    1. The geometrical weight :math:`w_{ls, \rm{geo}}` depends on lens and source redshift information. When considering only 
+    redshift point estimates, the weights read
+
+        .. math::
+
+            w_{ls, \rm{geo}} = \Sigma_c(\rm{cosmo}, z_L, z_{\rm src})^{-2}\;.
+    
+        If the redshift pdf of each source, :math:`p_{\rm photoz}(z_s)`, is known, the weights are computed instead as
+    
+        .. math::
+
+            w_{ls, \rm{geo}} = \left[\int_{\delta + z_L} dz_s p_{\rm photoz}(z_s) \Sigma_c(\rm{cosmo}, z_L, z_s)^{-1}\right]^2
+
+        for the tangential shear, the weights 'w_{ls, \rm{geo}}` are 1.
+
+    2. The shape weight :math:`w_{ls,{\rm shape}}` depends on shapenoise and/or shape measurement errors
+
+        .. math::
+
+            w_{ls, \rm{shape}} = 1/(\sigma_{\rm shapenoise}^2 + \sigma_{\rm measurement}^2)
+
+
+    3. The probability for a galaxy to be in the background of the cluster is defined by:
+
+        .. math::
+
+            P(z_s > z_l) = [\int_{z_L}^{+\infty} dz_s p_{\rm photoz}(z_s)
+
+        The function return the probability for a galaxy to be in the background of the cluster;
+        if photometric probability density functions are provoded, the function computes the above
+        integral. In the case of true redshifts, it returns 1 if :math:`z_s > z_l` else returns 0.
+
+    
+    Parameters
+    ----------
+    z_lens: float
+        Redshift of the lens.
+    z_source: array, optional
+        Redshift of the source. Used only with add_photoz=True.
+    cosmo: clmm.Comology object, None
+        CLMM Cosmology object.
+    pzpdf : array, optional
+        Photometric probablility density functions of the source galaxies
+    pzbins : array, optional
+        Redshift axis on which the individual photoz pdf is tabulated
+    shape_component1: array
+        The measured shear (or reduced shear or ellipticity) of the source galaxies
+    shape_component2: array
+        The measured shear (or reduced shear or ellipticity) of the source galaxies
+    shape_component1_err: array
+        The measurement error on the 1st-component of ellipticity of the source galaxies
+    shape_component2_err: array
+        The measurement error on the 2nd-component of ellipticity of the source galaxies
+    add_photoz : boolean
+        True if considering photometric redshifts
+    add_shapenoise : boolean
+        True for considering shapenoise in the weight computation
+    add_shape_error : boolean
+        True for considering measured shape error in the weight computation
+    is_deltasigma: boolean
+        Indicates whether it is the excess surface density or the tangential shear
+    validate_input: bool
+        Validade each input argument
+
+    Returns
+    -------
+    w_ls: array
+        Individual lens source pair weights
+    p_background : array
+        Probability for being a background galaxy
```

Comment:
It is confusing to have that function also returns the probability of being a background galaxy. This should be done in a separate function, e.g. `compute_background_probability`.